tests/real_call_test.py

```diff
@@ -28,25 +29,28 @@ def test_gets():
         ("status/204", 204, "", None),
     ]
 
-    with socketserver.TCPServer(("127.0.0.1", 5000), LocalTestServerHandler) as server:
-        server_thread = threading.Thread(target=server.serve_forever)
-        server_thread.daemon = True
-        server_thread.start()
+    for case in cases:
+        requests = adafruit_requests.Session(socket, ssl.create_default_context())
+        with requests.get(f"http://127.0.0.1:5000/{case[path_index]}") as response:
+            assert response.status_code == case[status_code_index]
+            if case[text_result_index] is not None:
+                assert response.text == case[text_result_index]
+            if case[json_keys_index] is not None:
+                for key, value in case[json_keys_index].items():
+                    assert response.json()[key] == value
 
-        time.sleep(2)  # Give the server some time to start
+        adafruit_connection_manager.connection_manager_close_all(release_references=True)
 
-        for case in cases:
-            requests = adafruit_requests.Session(socket, ssl.create_default_context())
-            with requests.get(f"http://127.0.0.1:5000/{case[path_index]}") as response:
-                assert response.status_code == case[status_code_index]
-                if case[text_result_index] is not None:
-                    assert response.text == case[text_result_index]
-                if case[json_keys_index] is not None:
-                    for key, value in case[json_keys_index].items():
-                        assert response.json()[key] == value
 
-            adafruit_connection_manager.connection_manager_close_all(release_references=True)
+def test_http_to_https_redirect():
+    url = "http://www.adafruit.com/api/quotes.php"
+    requests = adafruit_requests.Session(socket, ssl.create_default_context())
+    with requests.get(url) as response:
+        assert response.status_code == 200
```

Comment:
@FoamyGuy worth adding parametrize, to make sure that allow_redirects actually works as well?
```suggestion
@pytest.mark.parametrize(
    ("allow_redirects", "status_code"),
    (
        (True, 200),
        (False, 301),
    ),
)
def test_http_to_https_redirect(allow_redirects, status_code):
    url = "http://www.adafruit.com/api/quotes.php"
    requests = adafruit_requests.Session(socket, ssl.create_default_context())
    with requests.get(url, allow_redirects=allow_redirects) as response:
        assert response.status_code == status_code
```